pkg/driver/node.go

```diff
@@ -584,16 +584,19 @@ func (d *nodeService) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetC
 }
 
 func (d *nodeService) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
+	var segments map[string]string
 	klog.V(4).Infof("NodeGetInfo: called with args %+v", req)
 
-	in, err := d.cloud.GetPVMInstanceByID(d.pvmInstanceId)
+	in, err := d.cloud.GetPVMInstanceDetails(d.pvmInstanceId)
 	if err != nil {
 		klog.Errorf("failed to get the instance for pvmInstanceId %s, err: %s", d.pvmInstanceId, err)
 		return nil, fmt.Errorf("failed to get the instance for pvmInstanceId %s, err: %s", d.pvmInstanceId, err)
 	}
 
-	segments := map[string]string{
-		DiskTypeKey: in.DiskType,
+	if in != nil {
```

Comment:
Instead of sending empty storage type in segments can we handle this at line 591 to throw an error if instanceDetails is empty?